### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by maniac103**
With this I think we are pretty much feature complete with regards to OH binding support. Please give this a try on your T8 (should work with binding after adding T8 to the supported devices json file). For your 901, commands should hopefully work (please try!); the XML-over-MQTT event interface is still missing.

### CODE REVIEW

File: src/main/java/dev/pott/sucks/cleaner/ErrorDescription.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ErrorDescription {
+    public final int errorCode;
+    public final String description;
+
+    public ErrorDescription(int errorCode) {
+        this.errorCode = errorCode;
+        this.description = ERROR_DESCS.get(errorCode);
```
Comment (maniac103) [inline src/main/java/dev/pott/sucks/cleaner/ErrorDescription.java:14]: I'm still unsure what the best way to convey the error code to OH is. I see 3 options:
- Use only numbers: translation to string would be OH job (pro: no i18n issues, con: leaky abstraction)
- Use the current way or plain strings (exactly the other way around)
- Convert to an enum?

Thoughts?

Comment (JohannesPtaszyk) [reply src/main/java/dev/pott/sucks/cleaner/ErrorDescription.java:14]: Hmm, do we have an example on how others do it? 
I guess we are not the first to face this kind of situation. 
Could we have an enum for the error definition and map it to localized strings in OH?

**Review by JohannesPtaszyk — APPROVED** [review]

File: src/main/java/dev/pott/sucks/cleaner/ErrorDescription.java (hunk 1 above)
Comment (maniac103) [reply src/main/java/dev/pott/sucks/cleaner/ErrorDescription.java:14]: > Could we have an enum for the error definition and map it to localized strings in OH?

I guess so, but the disadvantage of that is that unknown error codes get lost. Maybe one idea would be a middle ground approach of using an `int` as error code and having constant definitions that map it to a meaning.

Other bindings sometimes map the numeric code right into a channel (e.g. irobot), but I find that approach somewhat ugly.

Comment (JohannesPtaszyk) [reply src/main/java/dev/pott/sucks/cleaner/ErrorDescription.java:14]: I think returning an error with enum + error code properties could be a good choice.

Comment (JohannesPtaszyk) [reply src/main/java/dev/pott/sucks/cleaner/ErrorDescription.java:14]: Could you maybe provide me with instructions on how to define the device in the json file? :) 
Really busy at the moment, so would be really glad about some basic steps. Thank you already!

Comment (maniac103) [reply src/main/java/dev/pott/sucks/cleaner/ErrorDescription.java:14]: I guess for the T8 you can copy the 950 entry for now; just the device class needs adaption. The correct value for that can be found in the device list cloud response.

Comment (JohannesPtaszyk) [reply src/main/java/dev/pott/sucks/cleaner/ErrorDescription.java:14]: Will check today :)
If you are okay with it I would also start moving over this repository implementation to the Openhab binding. 

One more thing I would like to do is to make the api-keys configurable from within openhab to easily provide new ones in the future.

Comment (maniac103) [reply src/main/java/dev/pott/sucks/cleaner/ErrorDescription.java:14]: >  If you are okay with it I would also start moving over this repository implementation to the Openhab binding.

[Done](https://github.com/maniac103/openhab2-addons/commit/96b4697390e218894fad7051f7cf2369cf663c5c) ;-)

> One more thing I would like to do is to make the api-keys configurable from within openhab to easily provide new ones in the future.

TBH, this seems like overkill to me as long as there's no signs of those keys needing replacement, as users won't know where to get the keys from anyway. I guess they're hardcoded in the app and were extracted from there.

**Comment by JohannesPtaszyk** [comment]
Thank you so much :) You put in so much effort! 
Tomorrow I will have time to test a ll the features with my  devices :) 
Maybe we can start providing the bidning to some testers soon

**Comment by maniac103** [comment]
After testing it on a T8/T9 I think doing a release in the forums makes sense, yes. Then we could have a look at supporting your 901 (shouldn't be too hard, just decoding the XML payloads via MQTT and testing the commands) and afterwards we should be good to go for a pull request towards OH.

**Pull request merged**